File: hackathon-backend/src/main/java/hackathon/Services/ScheduledTask/EmployeeService.java
package hackathon.Services.ScheduledTask;

import hackathon.Repository.EmployeeRepository;
import hackathon.client.MWLClient;
import hackathon.client.request.AboutMeTagsDTO;
import hackathon.entity.Employee;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.List;

@Singleton
public class EmployeeService {

    @Inject
    private MWLClient mwlClient;

    public void saveAllEmployeeAboutMeTagsDTO()
    {
        EmployeeRepository employeeRepository = new EmployeeRepository();
        List<AboutMeTagsDTO> aboutMeTagsDTOS = mwlClient.getAllEmployeeAboutMeTagsDTO();
        for(AboutMeTagsDTO aboutMeTagsDTO: aboutMeTagsDTOS){
            if(!employeeRepository.doesUserExist(aboutMeTagsDTO.getName())) {
                employeeRepository.save(aboutMeTagsDTO.getName(), aboutMeTagsDTO.getAboutMe(), aboutMeTagsDTO.getTags());
            }
        }
    }


}

File: hackathon-backend-kotlin/src/main/kotlin/hackathon/backend/kotlin/service/EmployeeService.kt
package hackathon.backend.kotlin.service

import hackathon.Repository.EmployeeRepository
import hackathon.backend.kotlin.client.MWLClient
import hackathon.backend.kotlin.client.request.AboutMeTagsDTO

import io.netty.handler.timeout.ReadTimeoutException
import javax.inject.Inject
import javax.inject.Singleton
import javax.transaction.Transactional


@Singleton
open class EmployeeService @Inject constructor(private val mwlClient: MWLClient,
private val employeeManager: EmployeeManager){

    open fun saveAllEmployeeAboutMeTagsDTO() {
        try {
            val aboutMeTagsDTOS: List<AboutMeTagsDTO>? = mwlClient.getAllEmployeeAboutMeTagsDTO()
            if (aboutMeTagsDTOS != null) {
                for (aboutMeTagsDTO in aboutMeTagsDTOS) {
                    if (!employeeManager.doesUserExist(aboutMeTagsDTO.name.orEmpty())
                    ) {
                        employeeManager.save(aboutMeTagsDTO.name, aboutMeTagsDTO.aboutMe, aboutMeTagsDTO.tags)
                    }
                }
            }
        } catch (e: ReadTimeoutException) {
            saveAllEmployeeAboutMeTagsDTO() //a timeout occurred try reading from MWL again
        }
    }
}
